Spark 2: Connection to Hive Metastore doesn't work when Hive Server 2 is running on the edge

The connection to the Hive Metastore from Spark does not work. We can see this by doing a simple `SHOW DATABASES` in `pyspark` shell:
```py
spark.sql('SHOW DATABASES')
```

With Spark 2, we have the following error:
```
py4j.protocol.Py4JJavaError: An error occurred while calling o37.sql.
...
Caused by: java.lang.NoClassDefFoundError: org/apache/tez/dag/api/TezConfiguration
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:661)
	at org.apache.hadoop.hive.ql.session.SessionState.start(SessionState.java:586)
	at org.apache.spark.sql.hive.client.HiveClientImpl.newState(HiveClientImpl.scala:180)
	at org.apache.spark.sql.hive.client.HiveClientImpl.<init>(HiveClientImpl.scala:114)
	... 41 more
Caused by: java.lang.ClassNotFoundException: org.apache.tez.dag.api.TezConfiguration
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at org.apache.spark.sql.hive.client.IsolatedClientLoader$$anon$1.doLoadClass(IsolatedClientLoader.scala:221)
	at org.apache.spark.sql.hive.client.IsolatedClientLoader$$anon$1.loadClass(IsolatedClientLoader.scala:210)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 45 more
```
Based on the warnings printed at instantiation of the `pyspark` shell for Spark 2, it seems that the Hive properties in `hive-site.xml` are not compatible with Hive 1.x (which is used by Spark 2):
```
2022-08-02 11:44:14,100 WARN conf.HiveConf: HiveConf of name hive.metastore.uri.selection does not exist
2022-08-02 11:44:14,101 WARN conf.HiveConf: HiveConf of name hive.metastore.truststore.path does not exist
2022-08-02 11:44:14,101 WARN conf.HiveConf: HiveConf of name hive.metastore.truststore.password does not exist
2022-08-02 11:44:14,101 WARN conf.HiveConf: HiveConf of name hive.metastore.use.SSL does not exist
2022-08-02 11:45:27,448 WARN conf.HiveConf: HiveConf of name hive.metastore.hmshandler.retry.attempts does not exist
2022-08-02 11:45:27,449 WARN conf.HiveConf: HiveConf of name hive.metastore.authentication does not exist
```
I think the error is coming both from incompatible `hive-site.xml` properties and missing Tez jars.

this message : `Caused by: java.lang.NoClassDefFoundError: org/apache/tez/dag/api/TezConfiguration` means that you are using the `hive-site.xml` of `hive-site.xml` of hive components and not the spark one

--- a/tdp_vars_defaults/spark/spark.yml
+++ b/tdp_vars_defaults/spark/spark.yml
@@ -153,6 +153,7 @@ hive_site_spark:
   hive.metastore.kerberos.principal: "hive_ms/_HOST@{{ realm }}"
   hive.metastore.execute.setugi: "true"
   hadoop.rpc.protection: AUTHENTICATION
+  hive.execution.engine: spark
 
 # Service restart policies
 spark_hs_restart: "no"

